pkg/gadgets/tcptracer/tracer/standard/tracer.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package standard
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/tcptracer/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/tcptracer/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+type Tracer struct {
+	gadgets.StandardTracerBase
+
+	resolver      containercollection.ContainerResolver
+	eventCallback func(types.Event)
+	node          string
+}
+
+func NewTracer(config *tracer.Config, resolver containercollection.ContainerResolver,
+	eventCallback func(types.Event), node string) (*Tracer, error,
+) {
+	lineCallback := func(line string) {
+		event := types.Event{}
+		event.Type = eventtypes.NORMAL
+
+		// "Hack" to avoid changing the BCC tool implementation
```

Comment:
There are two issues. The first one is that `ip` is misleading in the json message,  I think a more explicit `ipversion` makes it more clear. Regarding `type`, we can't use it because it's already used in the base event. 

I wanted to modified the BCC tools but I found out it requires a lot of changes so I decided to implement this approach, even it is not so clean it's easier.